webservices/load_legal_docs.py

```diff
@@ -247,3 +261,182 @@ def load_advisory_opinions_into_s3():
                 print("%d of %d advisory opinions written to s3" % (i + 1, len(new_docs)))
         else:
             print("No new advisory opinions found.")
+
+def process_mur_pdf(mur_no, pdf_key, bucket):
+    response = requests.get('http://www.fec.gov/disclosure_data/mur/%s.pdf'
+                            % mur_no, stream=True)
+
+    with NamedTemporaryFile('wb+') as pdf:
+        for chunk in response:
+            pdf.write(chunk)
+
+        pdf.seek(0)
+        try:
+            pdf_doc = slate.PDF(pdf)
+        except:
+            print('Could not parse pdf: %s' % pdf_key)
+            pdf_doc = []
+        pdf_pages = len(pdf_doc)
+        pdf_text = ' '.join(pdf_doc)
+        pdf_size = getsize(pdf.name)
+
+        pdf.seek(0)
+        bucket.put_object(Key=pdf_key, Body=pdf,
+                          ContentType='application/pdf', ACL='public-read')
+        return pdf_text, pdf_size, pdf_pages
+
+def get_subject_tree(html, tree=[]):
+    # get next token
+    root = re.match("([^<]+)(?:<br>)?(.*)", html, re.S)
+    list_item = re.match("<li>(.*?)</li>(.*)", html, re.S)
+    unordered_list = re.match("<ul\s+class='no-top-margin'>(.*)", html, re.S)
+    end_list = re.match("</ul>(.*)", html, re.S)
+    empty = re.match("\s*<br>\s*", html)
+
+    if empty:
+        pass
+    elif end_list:
+        # reduce
+        sub_tree = []
+        node = tree.pop()
+        while node != 'ul_start':
+            sub_tree.append(node)
+            node = tree.pop()
+
+        sub_tree.reverse()
+        if tree[-1] != 'ul_start':
+            tree[-1]['children'] = sub_tree
+        else:
+            tree.extend(sub_tree)
+    elif unordered_list:
+        # shift
+        tree.append('ul_start')
+    elif list_item or root:
+        # shift
+        subject = re.sub('\s+', ' ', (list_item or root).group(1))\
+                    .strip().lower().capitalize()
+        tree.append({'text': subject})
+    else:
+        print(html)
+        raise Exception("Could not parse next token.")
```

Comment:
This looks like a pretty standard parse switch, but if you're using a particular algorithm, or can describe the high-level idea of how the parsing works, I think that would be helpful to readers.